forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4506,7 +4511,6 @@ public final void removePerpetual(final long timestamp) {
         for (Map<String, Object> p : perpetual) {
             if (p.get("Timestamp").equals(timestamp)) {
                 toRemove = p;
-                break;
```

Comment:
why remove optimization?

Reply:
Some Perpetual effects have multiple entries with the same timestamp. This card for example has "Incorporate" and "Colors" categories with the same timestamp.

As I think about it more, there are further problems as Guildpact Greenwarden here has "Incorporate" and "Colors" with one timestamp, but "BoostPT" with another timestamp since that's in PumpEffect 😦

Reply:
well I think when you implemented this function it was for a very specific card?
if the code is supposed to handle multiple entries it would need updating anyway